### PR TITLE
add in better exception information for the REST API wrappers

### PR DESCRIPTION
**Pull request by yvchan**
Created a customized exception for REST errors so that we can identify what happened in the highlevels of code.

### CODE REVIEW

**Comment by ddebrunner** [comment]
Can one of the admins verify this patch?

**Comment by yvchan** [comment]
was just talking to John M about this code, and he's suggesting that i do not override getMessage() but set the message in the exception instead.  I'll make the change.

**Comment by yvchan** [comment]
ok, the update is here...  added in a factory method to handle getting the streams error response back ensuring the getMessage() method returns a human readable string when it's set.

File: java/src/com/ibm/streamsx/rest/RESTException.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        return status;
+    }
+
+    /**
+     * Gets the IBM Streams message id for this exception
+     * 
+     * @return {@link String}
+     */
+
+    public String getStreamsErrorMessageId() {
```
Comment (ddebrunner) [inline java/src/com/ibm/streamsx/rest/RESTException.java]: What's the purpose of this method?

File: java/src/com/ibm/streamsx/rest/RESTException.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        }
+        return id;
+    }
+
+    /**
+     * Gets the IBM Streams message for this exception
+     * 
+     * @return {@link String}
+     */
+    public String getStreamsErrorMessage() {
```
Comment (ddebrunner) [inline java/src/com/ibm/streamsx/rest/RESTException.java]: What's the purpose of this method?

Comment (yvchan) [inline java/src/com/ibm/streamsx/rest/RESTException.java]: i'll remove this, it isn't necessary.  a previous iteration did not set the message field the Exception, and this was to get the message from the error structure.  however, i re-worked this to always set the message, so this method is no longer required.

**Comment by yvchan** [comment]
ok, the method has been removed and this should be ready to merge.

**Pull request merged**